packages/imperative/src/config/cmd/secure/secure.handler.ts

```diff
@@ -50,7 +69,69 @@ export default class SecureHandler implements ICommandHandler {
             }
         }
 
+        if (authTokenProp != null) {
+            const propValue = await this.handlePromptForAuthToken(config, authTokenProp) ||
+                await params.response.console.prompt(`Please enter ${authTokenProp}: `, {hideText: true});
+
+            // Save the value in the config securely
+            if (propValue) {
+                config.set(authTokenProp, propValue, { secure: true });
+            }
+        }
+
         // Write the config layer
         await config.save(false);
     }
+
+    /**
+     * Checks if authentication service is associated with an auth token
+     * property. If an auth service is found, we log in to it obtain the token
+     * instead of prompting for the value.
+     * @param config Team config class from which the property was loaded
+     * @param propPath JSON property path of the auth token
+     * @returns Token value, or undefined if none was obtained
+     */
+    private async handlePromptForAuthToken(config: Config, propPath: string): Promise<string | undefined> {
+        const profilePath = propPath.slice(0, propPath.indexOf(".properties"));
+        const profileType = lodash.get(config.properties, `${profilePath}.type`);
+        const profile = config.api.profiles.get(profilePath.replace(/profiles\./g, ""));
+        if (profileType == null || profile.tokenType == null) {
+            return;  // Can't find auth service if profile type or token type are unknown
+        }
+
+        const authConfigs: ICommandProfileAuthConfig[] = [];
+        ImperativeConfig.instance.loadedConfig.profiles.forEach((profCfg) => {
+            if (profCfg.type === profileType && profCfg.authConfig != null) {
+                authConfigs.push(...profCfg.authConfig);
+            }
+        });
+
+        for (const authConfig of authConfigs) {
```

Comment:
Just a question: Is the for loop used to handle the case where multiple APIML servers might be configured?

Reply:
This implementation supports only one APIML login per team config file. So if a file contains multiple `tokenValue` properties associated with profiles that have `tokenType=apimlAuthenticationToken`, only the first occurrence would attempt to log in to APIML.

Imperative supports multiple auth configs which is the reason for this for loop, although Zowe CLI has only one auth config for base profiles.